lib/ansible/module_utils/gcp.py

```diff
@@ -231,7 +231,7 @@ def _validate_credentials_file(module, credentials_file, require_valid_json=True
     """
     try:
         # Try to read credentials as JSON
-        with open(credentials_file) as credentials:
+        with open(os.path.expanduser(credentials_file)) as credentials:
```

Comment:
this is the wrong fix, set the type=path in argspec and it will also resolve issus like $HOME/file.json